drivers/can/can_native_posix_linux.c

```diff
@@ -0,0 +1,435 @@
+/*
+ * Copyright (c) 2022 Martin Jäger <martin@libre.solar>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_native_posix_linux_can
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <zephyr/drivers/can.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/socketcan.h>
+#include <zephyr/net/socketcan_utils.h>
+
+#include "can_utils.h"
+#include "can_native_posix_linux_socketcan.h"
+
+LOG_MODULE_REGISTER(can_npl, CONFIG_CAN_LOG_LEVEL);
+
+struct can_filter_context {
+	can_rx_callback_t rx_cb;
+	void *cb_arg;
+	struct can_filter filter;
+};
+
+struct can_npl_data {
+	struct can_filter_context filters[CONFIG_CAN_MAX_FILTER];
+	struct k_mutex filter_mutex;
+	struct k_sem tx_idle;
+	struct k_sem tx_done;
+	can_tx_callback_t tx_callback;
+	void *tx_user_data;
+	bool loopback;
+	bool mode_fd;
+	int dev_fd; /* Linux socket file descriptor */
+	struct k_thread rx_thread;
+	K_KERNEL_STACK_MEMBER(rx_thread_stack, CONFIG_ARCH_POSIX_RECOMMENDED_STACK_SIZE);
```

Comment:
To fix the CI issue:

```suggestion
	struct k_thread rx_thread;

	K_KERNEL_STACK_MEMBER(rx_thread_stack, CONFIG_ARCH_POSIX_RECOMMENDED_STACK_SIZE);
```